platforms/core-runtime/service-provider/src/main/java/org/gradle/internal/service/CloseableServiceRegistry.java

```diff
@@ -18,8 +18,25 @@
 
 import java.io.Closeable;
 
+/**
+ * Managed version of a {@link ServiceRegistry} that controls the lifetime of the registry members,
+ * such as service instances and factories.
+ * <p>
+ * Members created by the registry are stopped and closed when this registry is {@link #close() closed}.
+ * <p>
+ * If a member implements {@link java.io.Closeable#close() Closeable} or
+ * {@link org.gradle.internal.concurrent.Stoppable#stop() Stoppable} then the appropriate
+ * method is called to dispose of it.
+ * <p>
+ * Members are closed in reverse dependency order.
+ */
 public interface CloseableServiceRegistry extends ServiceRegistry, Closeable {
 
+    /**
+     * Closes this registry by stopping and closing all members managed by it.
```

Comment:
You'd be surprised at how many places we cannot use the registries with the `try-resources` clause due to how lifecycles are structured. So, it makes sense to reiterate it here 👍 

The only thing that deterred me from doing so immediately was the Javadoc errors because `Stoppable` is not actually visible from this project.